pkg/remote/remote.go

```diff
@@ -120,13 +140,53 @@ func Parse(goGetterSrc string) (*Source, error) {
 	}, nil
 }
 
-func (r *Remote) Fetch(goGetterSrc string, cacheDirOpt ...string) (string, error) {
-	u, err := Parse(goGetterSrc)
+func ParseNormal(path string) (*Source, error) {
+	_, err := ParseNormalProtocol(path)
+	if err != nil {
+		return nil, err
+	}
+
+	u, err := url.Parse(path)
+	dir := filepath.Dir(u.Path)
+	if len(dir) > 0 {
+		dir = dir[1:]
+	}
+
+	return &Source{
+		Getter:   "normal",
+		User:     u.User.String(),
+		Scheme:   u.Scheme,
+		Host:     u.Host,
+		Dir:      dir,
+		File:     filepath.Base(u.Path),
+		RawQuery: u.RawQuery,
+	}, err
+}
+
+func ParseNormalProtocol(path string) (string, error) {
+	parts := strings.Split(path, "://")
+
+	if len(parts) == 0 {
+		return "", fmt.Errorf("failed to parse URL %s", path)
+	}
+	protocol := strings.ToLower(parts[0])
+
+	protocols := []string{"s3", "http", "https"}
+	for _, option := range protocols {
```

Comment:
we can use the go 1.21 slice package to judge if the `protocol` is in `protocols`.